crates/dbsp/src/trace/ord/mod.rs

```diff
@@ -34,3 +34,39 @@ pub use VecIndexedZSet as OrdIndexedZSet;
 pub use VecKeyBatch as OrdKeyBatch;
 pub use VecValBatch as OrdValBatch;
 pub use VecZSet as OrdZSet;
+
+use crate::DBData;
+use crate::DBWeight;
+
+use super::layers::OrdOffset;
+use super::Batch;
+use super::BatchReader;
+
+/// A batch that can be written to storage, by being transformed into an
+/// on-storage batch.
+///
+/// Stream operator
+/// [`Stream::spill`](crate::circuit::circuit_builder::Stream::spill) spills the
+/// batch to storage.
+pub trait SpillableBatch: BatchReader<Time = ()> {
```

Comment:
We should figure out a way not to specify this trait bound for every single operator.

Reply:
cc @blp

Reply:
It's really just for a few, the ones that actually create a trace in storage.

I'm happy to reduce or eliminate it later.